R-packages/covidcast/R/covidcast.R

```diff
@@ -549,30 +588,52 @@ covidcast_days <- function(data_source, signal, start_day, end_day, geo_type,
       message(summary)
     }
     if (dat[[i]]$message == "success") {
-      returned_geo_values <- dat[[i]]$epidata$geo_value
+      desired_geos <- tolower(unique(geo_value))
+      returned_epidata <- dat[[i]]$epidata
+      returned_geo_array <- returned_epidata %>%
+        select(geo_value, time_value) %>%
+        group_by(time_value) %>%
+        summarize(geo_value = list(geo_value))
+      returned_time_values <- returned_geo_array$time_value
+      if (length(returned_time_values) != length(time_values)) {
+        missing_time_values <- setdiff(time_values, returned_time_values)
+        missing_dates <- ymd(missing_time_values)
+        warn(sprintf("Data not fetched for the following days: %s",
+                     paste(missing_dates, collapse = ", ")),
+             data_source = data_source,
+             signal = signal,
+             day = missing_dates,
+             geo_value = geo_value,
+             api_msg = dat[[i]]$message,
+             class = "covidcast_missing_geo_values"
+        )
+      }
       if (!identical("*", geo_value)) {
-        missed_geos <- setdiff(tolower(geo_value),
-                               tolower(returned_geo_values))
-        if (length(missed_geos) > 0) {
-          missed_geos_str <- paste0(missed_geos, collapse = ", ")
-          warn(sprintf("Data not fetched for some geographies on %s: %s",
-                         query_day, missed_geos_str),
+        missing_geo_array <- returned_geo_array[
+          lapply(returned_geo_array$geo_value, length) < length(desired_geos), ]
+        if (nrow(missing_geo_array) > 0) {
+          missing_geo_array$warning <-
+            unlist(apply(returned_geo_array,
+                         1,
+                         FUN = function(row) geo_warning_message(row,
+                                                                 desired_geos)))
+          warn(missing_geo_array$warning,
                data_source = data_source,
                signal = signal,
-               day = query_day,
+               day = ymd(missing_geo_array$time_value),
                geo_value = geo_value,
-               api_msg = dat[[i]]$message,
-               class = "covidcast_missing_geo_values"
-               )
+               msg = dat[[i]]$message,
```

Comment:
This needs to go back to being called `api_msg`; maybe this changed back when you rebased on #252?

```suggestion
               api_msg = dat[[i]]$message,
```

Reply:
I had some merge conflicts early on, seems likely I grabbed the wrong line there. Fixed.